flink-tests/src/test/java/org/apache/flink/test/state/ChangelogRescalingITCase.java

```diff
@@ -329,24 +329,30 @@ public boolean request() {
     private String checkpointAndCancel(JobID jobID) throws Exception {
         waitForCheckpoint(jobID, cluster.getMiniCluster(), 1);
         cluster.getClusterClient().cancel(jobID).get();
-        checkStatus(jobID);
+        waitForSuccessfulTermination(jobID);
         return CommonTestUtils.getLatestCompletedCheckpointPath(jobID, cluster.getMiniCluster())
                 .<NoSuchElementException>orElseThrow(
                         () -> {
                             throw new NoSuchElementException("No checkpoint was created yet");
                         });
     }
 
-    private void checkStatus(JobID jobID) throws InterruptedException, ExecutionException {
-        if (cluster.getClusterClient().getJobStatus(jobID).get().isGloballyTerminalState()) {
-            cluster.getClusterClient()
-                    .requestJobResult(jobID)
-                    .get()
-                    .getSerializedThrowable()
-                    .ifPresent(
-                            serializedThrowable -> {
-                                throw new RuntimeException(serializedThrowable);
-                            });
+    private void waitForSuccessfulTermination(JobID jobID)
+            throws InterruptedException, ExecutionException {
+        while (true) {
```

Comment:
Maybe we can use `CommonTestUtils.waitUntilCondition` instead of `while loop`.

Reply:
Good point!